cilium/CFP-22215-mutual-auth-for-service-mesh.md

```diff
@@ -0,0 +1,455 @@
+
+Extended agent downtime could impact the ability to properly terminate session authentication for peers, depending on the datapath implementation. N-tuple match on connection endpoints in new auth map discusses how timestamps could be integrated into the datapath to ensure connection termination when mutual authentication reauthentication periods elapse. If these timestamps are not integrated into the datapath, then we would have to accept that authenticated sessions may remain authenticated indefinitely during agent downtime.
+
+### Key question: How dynamic should the cert granularity be?
+A few options for certificate granularity are described above, such as associating one cert with each node; one with each namespace; one with each deployment/identity. Should these be configurable based on the policy for individual endpoints, or statically defined across the cluster?
+
+#### Option 1: Define mesh-auth-mode via daemon config
+
+##### Pros
+Upper bound for auth connections / node is pre-defined
+##### Cons
+
+
+#### Option 2: Define mesh-auth-mode via CRD policy
+
+##### Pros
+More user control over the granularity. For high-connection pods like Ingress, it could help to bound the maximum number of auth connections per the discussion under Impact: Authentication connections per node.
+##### Cons
+More complicated
+Let's say two peers have two different configurations, one with node authentication and another with namespace. Does that mean that 2+ different auth sessions are established to provide the authentication the user expects?
+
+### Key question: How to implement dataplane auth check
+
+The dataplane authentication check in eBPF requires two components: One is a bit in the policy logic that marks that affected connections must be authenticated. This can be implemented with minimal changes by adding it into the existing policymap. The second is how we check that a connection is authenticated. This part must be informed by the userspace local control plane populating some state that is available to the eBPF dataplane.
+
+#### Chosen alternative: N-tuple match on connection endpoints in new auth map
+
+A new map is introduced into the eBPF dataplane that allows the packet-handling logic to look up for some subset of `{src, dst, …}` fields of the packet and confirm whether authentication has already occurred for this set of peers.
+Pros
+Allow multiple connections between the same peers to leverage the same authentication channel, rather than forcing every connection to individually authenticate
+Could support pre-authentication + pre-population (subject to scale concerns)
+Cons
+More expensive per-packet to look up an additional map using packet fields
+Additional memory overhead (~N MB)
```

Comment:
This is missing formatting?
```suggestion
*Pros*
- Allow multiple connections between the same peers to leverage the same authentication channel, rather than forcing every connection to individually authenticate
- Could support pre-authentication + pre-population (subject to scale concerns)
*Cons*
- More expensive per-packet to look up an additional map using packet fields
- Additional memory overhead (~N MB)
```